### PR TITLE
Fix ClassRenamePhpDocNodeVisitor with namespace check on auto import

### PR DESCRIPTION
**Pull request by samsonasik**
The `@noRector` removal by PR:

- https://github.com/rectorphp/rector-src/pull/3148

cause error in tests : 

```bash

There were 3 failures:

1) Rector\Tests\Renaming\Rector\Name\RenameClassRector\AutoImportNamesTest::test with data set #0 ('/home/runner/work/rector-src/...hp.inc')
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
     /**
      * @param DateTime $foo
      *
-     * @return Storage|\Illuminate\Support\Facades\Storage
+     * @return \Illuminate\Support\Facades\Storage|\Illuminate\Support\Facades\Storage
      */
     public function bar($foo){}
```

Ref https://github.com/rectorphp/rector-src/actions/runs/3608581525/jobs/6081273124#step:5:114

This PR try to fix it

### CODE REVIEW

**Comment by samsonasik** [comment]
Finally 🎉 All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules-tests/Renaming/Rector/Name/RenameClassRector/FixtureAutoImportNames/class_with_same_name_inside_different_namespace.php.inc [hunk 1]
```diff
@@ -24,7 +24,7 @@ class Foo2 extends FooBar
     /**
      * @param DateTime $foo
      *
-     * @return Storage|\Illuminate\Support\Facades\Storage
+     * @return Foo2\Storage|\Illuminate\Support\Facades\Storage
```
Comment (samsonasik) [inline rules-tests/Renaming/Rector/Name/RenameClassRector/FixtureAutoImportNames/class_with_same_name_inside_different_namespace.php.inc]: Ooops, this seems wrong, see https://3v4l.org/1qq8K

Comment (samsonasik) [reply rules-tests/Renaming/Rector/Name/RenameClassRector/FixtureAutoImportNames/class_with_same_name_inside_different_namespace.php.inc]: Fixed 🎉 https://github.com/rectorphp/rector-src/pull/3149/commits/211bf0e6f65817a91781a612e666ec13b33daf77

**Comment by samsonasik** [comment]
Finally 🎉 All checks have passed 🎉 @TomasVotruba I think it is ready.

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thanks 👍

**Pull request merged**